src/ServiceFabric/ServiceFabric/Commands/ManagedClusters/AddAzServiceFabricManagedNodeTypeVMExtension.cs

```diff
@@ -0,0 +1,165 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.ServiceFabric.Common;
+using Microsoft.Azure.Commands.ServiceFabric.Models;
+using Microsoft.Azure.Management.Internal.Resources;
+using Microsoft.Azure.Management.ServiceFabric;
+using Microsoft.Azure.Management.ServiceFabric.Models;
+
+namespace Microsoft.Azure.Commands.ServiceFabric.Commands
+{
+    [Cmdlet(VerbsCommon.Add, ResourceManager.Common.AzureRMConstants.AzurePrefix + Constants.ServiceFabricPrefix + "ManagedNodeTypeVMExtension", DefaultParameterSetName = ByObj, SupportsShouldProcess = true), OutputType(typeof(PSManagedNodeType))]
+    public class AddAzServiceFabricManagedNodeTypeVMExtension : ServiceFabricCommonCmdletBase
+    {
+        protected const string ByName = "ByName";
+        protected const string ByObj = "ByObj";
+
+        #region Params
+
+        #region Common params
+
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipelineByPropertyName = true, ParameterSetName = ByName,
+            HelpMessage = "Specify the name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty()]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, Position = 1, ValueFromPipelineByPropertyName = true, ParameterSetName = ByName,
+            HelpMessage = "Specify the name of the cluster.")]
+        [ResourceNameCompleter(Constants.ManagedClustersFullType, nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty()]
+        public string ClusterName { get; set; }
+
+        [Parameter(Mandatory = true, Position = 1, ValueFromPipelineByPropertyName = true, ParameterSetName = ByName,
+            HelpMessage = "Specify the name of the node type.")]
+        [ValidateNotNullOrEmpty()]
+        public string NodeTypeName { get; set; }
+
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ParameterSetName = ByObj,
+            HelpMessage = "Node Type resource")]
+        [ValidateNotNull]
+        public PSManagedNodeType InputObject { get; set; }
+
+        #endregion
+
+        [Parameter(Mandatory = true, HelpMessage = "extension name.")]
+        [Alias("ExtensionName")]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "If a value is provided and is different from the previous value, the extension handler will be forced to update even if the extension configuration has not changed.")]
+        public string ForceUpdateTag { get; set; }
```

Comment:
What is the sample input of ForceUpdateTag ?

Reply:
@msJinLei it can be any string. I have the same message that vmss has for this: https://docs.microsoft.com/en-us/powershell/module/az.Compute/Add-azVmssExtension?view=azps-4.6.1#parameters